pages/lobby.tsx

```diff
@@ -58,17 +63,47 @@ const SAMPLE_PLAYLIST_DATA = [
 ];
 
 function Lobby(): JSX.Element {
+  const {query} = useRouter();
+  const [lobbyData, setLobbyData] = useState<ILobbyData>(null);
+  const [loaded, setLoaded] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+
+  useEffect(() => {
+    async function loadData() {
+      const token = indexCookie('token');
+      if (!token || token === 'undefined' || query?.id == null) return;
+
+      const data = await fetchLobbyById(query.id as string, token);
+      setLobbyData(data);
+      setLoaded(true);
```

Comment:
i'm confused why you cant just do `!!lobbyData` to assess if it's loaded? if what ur looking for is the error code.. then maybe having just a separate state for the error code specifically? like if its invalid etc.